pkg/appliance/checks_test.go

```diff
@@ -57,25 +57,29 @@ increase the space on those appliances.
 		{
 			name: "warning version >= 18",
 			args: args{
-				stats: []openapi.StatsAppliancesListAllOfData{
+				stats: []openapi.ApplianceWithStatus{
 					{
-						Name:    openapi.PtrString("controller"),
-						Disk:    openapi.PtrFloat32(90),
-						Version: openapi.PtrString("5.5.4"),
-						DiskInfo: &openapi.StatsAppliancesListAllOfDiskInfo{
-							Total: openapi.PtrFloat32(100000000000),
-							Used:  openapi.PtrFloat32(90000000000),
-							Free:  openapi.PtrFloat32(10000000000),
+						Name:             "controller",
+						Disk:             openapi.PtrFloat32(90),
+						ApplianceVersion: openapi.PtrString("5.5.4"),
+						Details: &openapi.ApplianceWithStatusAllOfDetails{
+							Disk: &openapi.SystemInfo{
+								Total: openapi.PtrInt64(int64(100000000000)),
+								Used:  openapi.PtrInt64(int64(90000000000)),
+								Free:  openapi.PtrInt64(int64(10000000000)),
+							},
```

Comment:
I think this is the original issue in the API spec where you wanted int64 which led you to this journey of tasks. Thanks for taking this on!